src/core/dbus-rdt.c

```diff
@@ -0,0 +1,79 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "bus-util.h"
+#include "unit.h"
+#include "dbus-unit.h"
+#include "string-util.h"
+#include "dbus-rdt.h"
+#include "resctrl.h"
+#include "rdt.h"
+
+const sd_bus_vtable bus_rdt_vtable[] = {
+        SD_BUS_VTABLE_START(0),
+        SD_BUS_PROPERTY("L3CacheAllocationSize", "t", NULL, offsetof(Rdt, l3_size), 0),
+        SD_BUS_PROPERTY("L3CacheAllocationIds", "s", NULL, offsetof(Rdt, l3_ids), 0),
+        SD_BUS_VTABLE_END
+};
+
+int bus_rdt_set_property(
+                Unit *u,
+                const char *name,
+                sd_bus_message *message,
+                UnitWriteFlags flags,
+                sd_bus_error *error) {
+
+        Rdt *t = RDT(u);
+        int r;
+
+        assert(name);
+        assert(message);
+        assert(u);
+
+        flags |= UNIT_PRIVATE;
+
+        if (streq(name, "L3CacheAllocationSize")) {
+                uint64_t size;
+
+                r = sd_bus_message_read(message, "t", &size);
+                if (r < 0)
+                        return r;
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                        t->l3_size = size;
+                        t->l3_updated = true;
+                        unit_write_settingf(u, flags, name, "%s=%" PRIu64, name, size);
+                }
+
+                return 1;
+        } else if (streq(name, "L3CacheAllocationIds")) {
+                const char *ids = NULL;
+
+                r = sd_bus_message_read(message, "s", &ids);
+                if (r < 0)
+                        return r;
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                        char *buf = NULL;
```

Comment:
Please don't do initalizations where they are not needed (variables declared with `_cleanup_` are an exception).